include/trieste/parse.h

```diff
@@ -533,13 +534,13 @@ namespace trieste
   inline detail::Rule
   operator>>(detail::Located<const std::string&> s, detail::ParseEffect effect)
   {
-    return std::make_shared<detail::RuleDef>(s, effect);
+    return intrusive_ptr(new detail::RuleDef(s, effect));
```

Comment:
I don't like this pattern.  Raw `new` is generally considered bad form.  Could we wrap is and expose something like:
```C++
template<typename T, typename... Args>
intrusive_ptr<T> intrusive_refcounted<T>::make(Args&&... args) {
   return intrusive_ptr(new T(std::forward<Args>(args)...);
}
```

Or a `make_intrusive` function if the template and inheritance don't work out with it being a method.

Reply:
I got intrusive_ptr<T>::make working.

I could use it _almost_ everywhere, except where it was never possible due to NodeDef's private constructors.